v-next/hardhat-node-test-coverage/src/collector.ts

```diff
@@ -0,0 +1,44 @@
+import type { CoverageReport } from "hardhat/types/coverage";
+import type { NetworkHooks } from "hardhat/types/hooks";
+
+import { randomUUID } from "node:crypto";
+import path from "node:path";
+import { after, before } from "node:test";
+
+import { assertHardhatInvariant } from "@nomicfoundation/hardhat-errors";
+import { writeJsonFile } from "@nomicfoundation/hardhat-utils/fs";
+import hre from "hardhat";
+
+const report: CoverageReport = {
+  markerIds: [],
+};
+
+before(() => {
+  const networkHooks: Partial<NetworkHooks> = {
+    newConnection: async (
+      context,
+      networkName,
+      chainType,
+      networkConfigOverride,
+      next,
+    ) => {
+      // TODO: Define networkConfigOverride.observability.codeCoverage.onCollectedCoverageCallback
+      // if this is an EDR network config
+      return next(context, networkName, chainType, networkConfigOverride);
+    },
+  };
+
+  hre.hooks.registerHandlers("network", networkHooks);
```

Comment:
I thought this was going to be done in the `coverage` builtin plugin, with a non-dynamic handler, that would be a `nop` if `globalOptions.coverage` is false.

Reply:
The issue is that in the context of node:test worker, the `globalOptions.coverage` is currently `false`. We could enable it by setting `HARDHAT_COVERAGE=true` env var for the worker which would ensure the global option is set correctly.

Reply:
I addressed the global options in a subprocess issue here: https://github.com/NomicFoundation/hardhat/blob/6214a39955f6ff99c1c996c015b096759a011370/v-next/hardhat-node-test-runner/src/task-action.ts#L82

As for moving handlers registration to the coverage plugin, I think we should wait with that until the final decision on the EDR API. 

I also made some observations on the condition under which we could move the entire collection logic to the coverage plugin. See https://github.com/NomicFoundation/hardhat/pull/6595/commits/47ab90c8be85071c1a84752e5c100ab7fcfd8f0d

Reply:
I did move the logic to the coverage plugin in https://github.com/NomicFoundation/hardhat/pull/6595/commits/a37b7dd07e58ea19294e564058f45916b8cafe47 and I like how tiny the collector import on the node test worker side has become.